### ISSUE TITLE
stdlib is large

### ISSUE THREAD

**Issue by mustafa0x** [fasiha/ebisu.js#2]
stdlib, which is included for `betaln`, is causing my bundle size to blow up.

Shouldn't this be sufficient? https://github.com/substack/gamma.js/blob/master/index.js

**Comment by fasiha** [comment]
Hey thanks for writing! What bundler are you using? And how big is your bundle getting?

Browserify spits out a 340KB JS file, which Google Closure Compiler distills to a 76KB minified file. These are included in this repo: https://github.com/fasiha/ebisu.js/tree/gh-pages/dist

Are you seeing bundle size deltas much bigger than this? Or is even this large compared to the rest of your app?

**Comment by mustafa0x** [comment]
Thank you for making this!

> Browserify spits out a 340KB JS file, which Google Closure Compiler distills to a 76KB minified file.

Yep, I'm getting the same numbers here.

> Or is even this large compared to the rest of your app?

It is — my minified bundle is ~40kb.

Edit: I try my best to keep my bundle size small, in light of https://v8.dev/blog/cost-of-javascript-2019.

**Comment by fasiha** [comment]
Thanks for the info. Stdlib.js is very thorough so its `betaln` pulls in typed array implementations and replacements for some functions in `Math`, which we like when doing high-quality numerical analysis but I agree we should try to do better.

Substack's gamma.js is great: it passes my unit tests as is (with a bit more error than Stdlib, or [Cephes](https://github.com/nearform/node-cephes), another library I tried) and it results in a 12kb minified file. Thank you for bringing it to my attention!

If you are just targeting ES6 browsers, I made an ES6-targeted minified file that is 5kb.

Let me know if you see any irregularities with version 1.0.1 that I just npm-published. Gamma.js uses only one method, whereas Stdlib.js and Cephes use more than one method, picking the best one given the value of the argument—and their `betaln` are much more complex than just taking ratios of gammas as I do now. My unit tests cover a fair range of Ebisu models but I wouldn't be surprised if some users encounter numerical issues due to this. Until then though I'm happy to enjoy tiny bundles.

Thank you!
